File: src/main/java/com/risazhrn/ecommerce/entity/UserProduct.java
package com.risazhrn.ecommerce.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;


@Data
@AllArgsConstructor
@NoArgsConstructor
public class UserProduct {
    private Integer id;
    private Integer userId;
    private Integer productId;
    private Integer quantity;
    private Date createdAt;
    private Date updatedAt;

    @Data
    @AllArgsConstructor
    @NoArgsConstructor
    public static class UserListTransaction {
        private Integer id;
        private Integer userId;
        private Integer productId;
        private Integer quantity;
        private Integer price;
        private Date createdAt;
        private String userName;
        private String productName;
        private String urlImage;
        private String description;
        private String categoryName;
        private String address;
        private String phone;
    }

}

File: frontend/src/app/model/userProduct.ts
export class UserProduct{
    id!: number;
    userId!: number;
    productId!: number;
    quantity!: number;
}

export class UserProductInput{
    user_id!: number;
    product_id!: number;
    quantity!: number;
}

export class ListUserTransaction{
    id!: number;
    userId!: number;
    productId!: number;
    quantity!: number;
    userName!: string;
    productName!: string;
    createdAt!: Date;
    urlImage!: string;
}

export class DetailTransaction{
    id!: number;
    userId!: number;
    productId!: number;
    quantity!: number;
    productName!: string;
    createdAt!: Date;
    urlImage!: string;
    categoryName!: string;
    description!: string;
    userName!: string;
    price!: number;
    address!: string;
    phone!: string;
}